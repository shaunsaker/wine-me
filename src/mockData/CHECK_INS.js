import CHECK_IN from './CHECK_IN';

const CHECK_INS = {
  1: CHECK_IN,
};

export default CHECK_INS;
