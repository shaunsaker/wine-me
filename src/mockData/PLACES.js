import PLACE from './PLACE';

const PLACES = { 1: PLACE };

export default PLACES;
