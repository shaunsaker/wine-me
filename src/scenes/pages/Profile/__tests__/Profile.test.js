import React from 'react';
import renderer from 'react-test-renderer';

import { Profile } from '..';
import CHECK_INS from '../../../../mockData/CHECK_INS';
import PLACES from '../../../../mockData/PLACES';

jest.mock('../../../../components/PlaceList', () => 'PlaceList');
jest.mock('../../../../components/TabBar', () => 'TabBar');

describe('Profile', () => {
  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<Profile userCheckIns={CHECK_INS} places={PLACES} />);

      expect(component).toMatchSnapshot();
    });

    it('renders the blank state', () => {
      const component = renderer.create(<Profile userCheckIns={{}} places={PLACES} />);

      expect(component).toMatchSnapshot();
    });
  });
});
