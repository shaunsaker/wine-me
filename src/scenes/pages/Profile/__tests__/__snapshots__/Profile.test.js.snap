// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders renders the blank state 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingBottom": 0,
        "paddingTop": 22,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 100,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 22,
          },
          Object {
            "backgroundColor": "#48466D",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flex": 1,
        "justifyContent": "center",
        "maxWidth": 300,
        "paddingVertical": 100,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "#3D84A8",
            "fontSize": 64,
            "marginBottom": 10,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#212121",
          "fontFamily": "Nunito-SemiBold",
          "fontSize": 24,
          "marginBottom": 10,
          "textAlign": "center",
        }
      }
    >
      Turn water into wine.
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#212121",
          "fontFamily": "Nunito-Regular",
          "fontSize": 17,
          "textAlign": "center",
        }
      }
    >
      Start visiting wine farms, 'Check in' and they'll show up here.
    </Text>
  </View>
  <TabBar />
</View>
`;

exports[`Profile renders renders with minimum required props 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingBottom": 0,
        "paddingTop": 22,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 100,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 22,
          },
          Object {
            "backgroundColor": "#48466D",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginHorizontal": 15,
          "marginVertical": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#212121",
            "fontFamily": "Nunito-SemiBold",
            "fontSize": 17,
          }
        }
      >
        My places
      </Text>
    </View>
    <PlaceList
      data={
        Array [
          Object {
            "address": "Stellenbosch, South Africa",
            "id": "1",
            "location": Object {
              "lat": -33.9263749,
              "lng": 18.9319848,
            },
            "name": "Bartinney Tasting Shed",
            "openingHours": Array [
              "Monday: 12:00 – 5:00 PM",
              "Tuesday: 12:00 – 5:00 PM",
              "Wednesday: 12:00 – 5:00 PM",
              "Thursday: 12:00 – 5:00 PM",
              "Friday: 12:00 – 8:00 PM",
              "Saturday: 10:00 AM – 12:30 PM",
              "Sunday: Closed",
            ],
            "phoneNumber": "021 885 1013",
            "photoReferences": Array [
              "CmRaAAAAF5bkjtj7bFEGaZqTN1_uZrJMvM6EEOFle3UIDd5JEbZwb281f2F4EH-XR1o6frihMiOzrJEx1KlczmV8o4qLrYygDZE-TKthSRT1SdzvrEQ6AjQbCG6sVeZBR48V0lMLEhDkpyqJh9O9z4Ai-7All7swGhQwTkHxJ77yIZ9d0c8oqWifUrL6eA",
            ],
            "website": "http://www.bartinney.co.za",
          },
        ]
      }
    />
  </View>
  <TabBar />
</View>
`;
